packages/dev/sharedUiComponents/src/nodeGraphSystem/searchBox.tsx

```diff
@@ -0,0 +1,107 @@
+import * as React from "react";
+import type { StateManager } from "./stateManager";
+import "./searchBox.scss";
+import { NodeLedger } from "./nodeLedger";
+
+export interface ISearchBoxComponentProps {
+    stateManager: StateManager;
+}
+
+/**
+ * The search box component.
+ */
+export class SearchBoxComponent extends React.Component<ISearchBoxComponentProps, {isVisible: boolean, filter: string}> {
+    private _handleEscKey: (evt: KeyboardEvent) => void;
+    private _targetX: number;
+    private _targetY: number;
+    private _nodes: string[];
+
+    constructor(props: ISearchBoxComponentProps) {
+        super(props);
+
+        this.state = { isVisible: false, filter: "" };
+
+        this._handleEscKey = (evt: KeyboardEvent) => {
+            if (evt.key === "Escape") {
+                this.hide();
+            }
+        };
+
+        this.props.stateManager.onSearchBoxRequiredObservable.add((loc) => {
+            this._targetX = loc.x;
+            this._targetY = loc.y;
+            this.setState({ isVisible: true, filter: "" });
+            this.props.stateManager.hostDocument!.addEventListener("keydown", this._handleEscKey);
+        });
+    }
+
+    hide() {
+        this.setState({ isVisible: false });
+        this.props.stateManager.modalIsDisplayed = false;
+        this.props.stateManager.hostDocument!.removeEventListener("keydown", this._handleEscKey);

```

Comment:
Should we not remove on unmount as well just to be safe ?

Reply:
The component is static in the page. It does not get created manually